crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -570,6 +570,42 @@ where
     }
 }
 
+/// The [`Command`] type for unregistering one-shot systems from [`Commands`](crate::system::Commands).
+pub struct UnregisterSystemCached<I, O, M, S>
+where
+    I: SystemInput + 'static,
+    S: IntoSystem<I, O, M> + Send + 'static,
+{
+    system: S,
```

Comment:
Technically `world.unregister_system_cached` doesn't actually use the system argument passed into it, so this field could be removed. We could add a private `World::unregister_system_cached_inner::<S>` method that `World::unregister_system_cached` and this command both call into, using `S` directly instead of passing in a value of type `S`.

This isn't a big deal though so I'm not blocking on it.